File: marketmotormicroservices/producto-service/src/main/java/com/jma/productoservice/api/EmpleadoResponse.java
package com.jma.productoservice.api;

import com.jma.productoservice.dto.EmpleadoDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class EmpleadoResponse {

    private List<EmpleadoDto> content;
    private int pageNo;
    private int pageSize;
    private long totalElements;
    private int totalPages;
    private boolean last;


}

File: frontend-marketmotor/src/app/models/responseapi/EmpleadoResponse.ts
import { Empleado } from "../dtos/Empleado";

export class EmpleadoResponse{

    content: Empleado[];
    pageNo: number;
    pageSize: number;
    totalElements: number;
    totalPages: number;
    last: boolean;

    constructor(){
        this.content  = [new Empleado()]
        this.pageNo = 0;
        this.pageSize = 0;
        this.totalElements = 0;
        this.totalPages = 0;
        this.last = false;
    }

}
